sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java

```diff
@@ -146,10 +205,13 @@ public void deleteLogs(@NonNull String group, @NonNull String id) {
         AppCenterLog.debug(LOG_TAG, "The IDs for deleting log(s) is/are:");
 
         List<Long> dbIdentifiers = mPendingDbIdentifiersGroups.remove(group + id);
+        File directory = getLargePayloadGroupDirectory(group);
         if (dbIdentifiers != null) {
             for (Long dbIdentifier : dbIdentifiers) {
                 AppCenterLog.debug(LOG_TAG, "\t" + dbIdentifier);
                 mDatabaseStorage.delete(dbIdentifier);
+                //noinspection ResultOfMethodCallIgnored we are not checking SQLite delete either.
+                getLargePayloadFile(directory, dbIdentifier).delete();
```

Comment:
Could you make the identifier deleted after the file? If the identifier is deleted but something goes wrong while or before deleting the large file, there will be a large file leak.

Reply:
Was thinking about that this morning :D Will do.